### PR TITLE
Replaced pow with a multiply

### LINKED ISSUE #16136: Improve performance by replacing pow with multiply
While looking through the C code, I noticed a [reference](https://github.com/astropy/astropy/blob/47a8dac8f1dab6ce592be6048f70140adead4e1d/astropy/stats/src/compute_bounds.c#L45-L47) to `pow` within a `for` loop. `pow` is a slow operation meant for floating point exponents. Since the exponent here is integral (`==2`), this instance of `pow` should be replaced with a multiply. 

Existing code is:

```c
      for (i = 0; i < count; i++) {
        std += pow(mean - data_buffer[i], 2);
      }
```

Better performant code would be:

```c
      for (i = 0; i < count; i++) {
        const double dx = mean - data_buffer[i];
        std += dx * dx;
      }
```

### PR DESCRIPTION
**Pull request by manodeep**
### Description

This pull request is to replace the slow `pow` operation with a multiply when the relevant operation is squaring. 

Fixes #16136 

Not sure if this needs a change log entry. 

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/stats/src/compute_bounds.c [hunk 1]
```diff
@@ -43,7 +43,7 @@ void compute_sigma_clipped_bounds(double data_buffer[], int count, int use_media
 
       std = 0;
       for (i = 0; i < count; i++) {
-        std += pow(mean - data_buffer[i], 2);
+        std += (data_buffer[i] - mean) * (data_buffer[i] - mean);
```
Comment (larrybradley) [inline astropy/stats/src/compute_bounds.c:46]: Would it be more performant to assign `data_buffer[i] - mean` to a variable (e.g., `x`) and using `x * x`?

**Comment by pllim** [comment]
tl;dr on the original issue. Do you have a benchmark done for this PR? (Sorry, the auto benchmark PR is not merged yet.)

If there is significant speed up, probably want to advertise in change log?

**Comment by MridulS** [comment]
Fwiw I did do some local benchmarks. Didnt really make much of a difference.

File: astropy/stats/src/compute_bounds.c (hunk 1 above)
Comment (manodeep) [reply astropy/stats/src/compute_bounds.c:46]: Unlikely to be the case, since the compiler should put the result of the subtraction into a register and then multiply the contents of the register and that's effectively the same as saving into a separate variable (with local scope).

**Comment by manodeep** [comment]
> tl;dr on the original issue. Do you have a benchmark done for this PR? (Sorry, the auto benchmark PR is not merged yet.)
> 
> If there is significant speed up, probably want to advertise in change log?

As @\MridulS said, there is no speed-up because the compiler (with any reasonable optimization flag, say `-O2` set) replaces the `pow` with the corresponding multiply. But @\mhvk's suggestion (and I agree) was to replace the `pow` anyway just so no one has to ever think about this again.

**Review by larrybradley — APPROVED** [review]
Thanks, @manodeep
